examples/plot_learning_curve_p300.py

```diff
@@ -0,0 +1,139 @@
+"""
+===========================
+Within Session P300
+===========================
+
+This Example shows how to perform a within session analysis while also
+creating learning curves for a P300 dataset.
+
+We will compare three pipelines :
+
+- Riemannian Geometry
+- Jumping Means based Linear Discriminant Analysis
+- Time-Decoupled Linear Discriminant Analysis
+
+We will use the P300 paradigm, which uses the AUC as metric.
+"""
+# Authors: Jan Sosulski
+#
+# License: BSD (3-clause)
+
+from sklearn.pipeline import make_pipeline
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis as LDA
+from pyriemann.tangentspace import TangentSpace
+from pyriemann.estimation import XdawnCovariances
+from moabb.evaluations import WithinSessionEvaluationIncreasingData
+from moabb.paradigms import P300
+from moabb.datasets import BNCI2014009
+import moabb
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+import time
+import warnings
+from tdlda import Vectorizer as JumpingMeansVectorizer
+from tdlda import TimeDecoupledLda as TDLDA
```

Comment:
The example is very nice, but examples (especially those introducing a feature) should be reproducible without requiring to download any external code (other than MOABB or requirements.txt)

That being said, I found it very interesting to have advanced examples relying on external open source code, like you did. 

Could you make 2 examples from your code?
- one simple example with just RG (no JM + LDA/TD-LDA)
-  one advanced with the Jumping Means and Time Decoupled LDA, where you could provide further explanation on your method, pointing to your papers and indicating how to install the code.

Reply:
I agree, and I think having two examples is a very good idea.
I just figured `pyRiemann` is an external dependency as well, as it is only needed for the examples. Maybe the requirements could be split into requirements_core and requirements_extra or something to indicate what you would need to install when you  just want to use moabb vs running the examples.
I also have some additional points that we could discuss in the moabb office hours next week. (Thursday evening unfortunately does not work for me this week. Most weeks actually.)